applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_sink.c

```diff
@@ -81,6 +80,48 @@ static le_audio_receive_cb receive_cb;
 static bool init_routine_completed;
 static bool paused;
 
+static struct bt_csip_set_member_svc_inst *csip;
+/* Advertising data for peer connection */
+static uint8_t csip_rsi_adv_data[BT_CSIP_RSI_SIZE];
+
+#define CSIP_SET_SIZE 2
+enum csip_set_rank {
+	CSIP_HL_RANK = 1,
+	CSIP_HR_RANK = 2
+};
+
+/* Callback for locking state change from server side */
+static void csip_lock_changed_cb(struct bt_conn *conn, struct bt_csip_set_member_svc_inst *csip,
+				 bool locked)
+{
+	LOG_DBG("Client %p %s the lock", (void *)conn, locked ? "locked" : "released");
+}
+
+/* Callback for SIRK read request from peer side */
+static uint8_t sirk_read_req_cb(struct bt_conn *conn, struct bt_csip_set_member_svc_inst *csip)
+{
+	/* Accept the request to read the SIRK, but return encrypted SIRK instead of plaintext */
+	return BT_CSIP_READ_SIRK_REQ_RSP_ACCEPT_ENC;
```

Comment:
Why would this not be accepted? Do we need to check for encryption or other issues first?

Reply:
This shows our device supports encryped SIRK, and we won't let central to read the SIRK in plaintext.